lib/winstore.js

```diff
@@ -210,3 +215,145 @@ function launch(appId, options, callback) {
+ *
+ * @emits module:windowsphone#detected
+ * @emits module:windowsphone#error
+ *
+ * @returns {EventEmitter}
+ */
+function detect(options, callback) {
+	return magik(options, callback, function (emitter, options, callback) {
+		if (detectCache && !options.bypassCache) {
+			emitter.emit('detected', detectCache);
+			return callback(null, detectCache);
+		}
+
+		var results = {
+				windows: {},
+				issues: []
+			},
+			searchPaths = [
+				'HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Microsoft SDKs\\Windows', // probably nothing here
+				'HKEY_LOCAL_MACHINE\\Software\\Wow6432Node\\Microsoft\\Microsoft SDKs\\Windows' // this is most likely where Windows SDK will be found
+			];
+
+		function finalize() {
+			detectCache = results;
+			emitter.emit('detected', results);
+			callback(null, results);
+		}
+
+		async.each(searchPaths, function (keyPath, next) {
+			appc.subprocess.run('reg', ['query', keyPath], function (code, out, err) {
+				var keyRegExp = /.+\\(v\d+\.\d)$/;
+				if (!code) {
+					out.trim().split(/\r\n|\n/).forEach(function (key) {
+						key = key.trim();
+						var m = key.match(keyRegExp);
+						if (!m) {
+							return;
+						}
+						var version = m[1].replace(/^v/, '');
+						if (m) {
+							results.windows || (results.windows = {});
+							results.windows[version] = {
+								version: version,
+								registryKey: keyPath + '\\' + m[1],
+								supported: !options.supportedWindowsSDKVersions || appc.version.satisfies(version, options.supportedWindowsSDKVersions, false), // no maybes
+								path: null,
+								signTool: null,
+								makeCert: null,
+								pvk2pfx: null,
+								selected: false
+							};
+						}
+					});
+				}
+				next();
+			});
+		}, function () {
+			// check if we didn't find any Windows SDKs, then we're done
+			if (!Object.keys(results.windows).length) {
+				results.issues.push({
+					id: 'WINDOWS_STORE_SDK_NOT_INSTALLED',
+					type: 'error',
+					message: __('Microsoft Windows Store SDK not found.') + '\n' +
+						__('You will be unable to build Windows Store apps.')
+				});
+				return finalize();
+			}
+
+			// fetch Windows SDK install information
+			async.each(Object.keys(results.windows), function (ver, next) {
+				appc.subprocess.run('reg', ['query', results.windows[ver].registryKey, '/v', '*'], function (code, out, err) {
+					if (code) {
+						// bad key? either way, remove this version
+						delete results.windows[ver];
+					} else {
+						// get only the values we are interested in
+						out.trim().split(/\r\n|\n/).forEach(function (line) {
+							var parts = line.trim().split('   ').map(function (p) { return p.trim(); });
+							if (parts.length == 3) {
+								if (parts[0] == 'InstallationFolder') {
+									results.windows[ver].path = parts[2];
+
+									function addIfExists(key, exe) {
+										// TODO: There are also x64 and arm versions of the following...

```

Comment:
Hmm... I wonder how we should expose the different architecture versions? I suppose something like the following:

    results.windows['8.1']['arm'].signTool